src/ocf_freelist.c

```diff
@@ -397,7 +397,8 @@ ocf_freelist_t ocf_freelist_init(struct ocf_cache *cache)
 		env_atomic64_set(&freelist->part[i].curr_size, 0);
```

Comment:
I suggest to change the code so that return code from env_spinlock_init is propagated up:

line 392 should look like this:
result =env_spinlock_init(&freelist->lock[i])
if (result)
    goto spinlock_err;

Reply:
line 388:
if (!freelist->lock || !freelist->part) {
   result = -OCF_ERR_NO_MEM;
   goto free_allocs;
}